WeakAuras/GenericTrigger.lua

```diff
@@ -3472,66 +3492,95 @@ function WeakAuras.WatchUnitChange(unit)
+          if watchUnitChange.trackedUnits[unit] then
+            unitUpdate(unit, eventsToSend)
+            markerInit(unit)
+            reactionInit(unit)
+          end
         end
       elseif event == "PLAYER_TARGET_CHANGED" then
+        if not watchUnitChange.trackedUnits["target"] then
+          return
+        end
         unitUpdate("target", eventsToSend)
-        markerUpdate("target", eventsToSend)
-        reactionUpdate("target", eventsToSend)
+        markerInit("target")
+        reactionInit("target")
       elseif event == "PLAYER_FOCUS_CHANGED" then
+        if not watchUnitChange.trackedUnits["focus"] then
+          return
+        end
         unitUpdate("focus", eventsToSend)
-        markerUpdate("focus", eventsToSend)
-        reactionUpdate("focus", eventsToSend)
+        markerInit("focus")
+        reactionInit("focus")
       elseif event == "PLAYER_SOFT_ENEMY_CHANGED" then
+        if not watchUnitChange.trackedUnits["softenemy"] then
+          return
+        end
         unitUpdate("softenemy", eventsToSend)
-        markerUpdate("softenemy", eventsToSend)
-        reactionUpdate("softenemy", eventsToSend)
+        markerInit("softenemy")
+        reactionInit("softenemy")
       elseif event == "PLAYER_SOFT_FRIEND_CHANGED" then
+        if not watchUnitChange.trackedUnits["softfriend"] then
+          return
+        end
         unitUpdate("softfriend", eventsToSend)
-        markerUpdate("softfriend", eventsToSend)
-        reactionUpdate("softfriend", eventsToSend)
+        markerInit("softfriend")
+        reactionInit("softfriend")
       elseif event == "RAID_TARGET_UPDATE" then
         for unit in pairs(watchUnitChange.raidmark) do
           markerUpdate(unit, eventsToSend)
         end
       elseif event == "UNIT_FACTION" then
-        if unit:sub(1, 9) == "nameplate" then
+        if watchUnitChange.trackedUnits[unit] then
           reactionUpdate(unit, eventsToSend)
         end
       elseif event == "UNIT_PET" then
         local pet = WeakAuras.unitToPetUnit[unit]
-        if pet then
+        if pet and watchUnitChange.trackedUnits[pet] then
           eventsToSend["UNIT_CHANGED_" .. pet] = pet
         end
       elseif event == "PLAYER_ROLES_ASSIGNED" then
+        if not watchUnitChange.trackedUnits[unit] then
+          return
+        end
         roleUpdate(unit, eventsToSend)
       elseif event == "UNIT_TARGET" then
-        -- i dont know why this event is register
+        local unitTarget = unit .. "target"
+        if not watchUnitChange.trackedUnits[unitTarget] then
+          return
+        end
+        unitUpdate(unitTarget, eventsToSend)
+        markerInit(unitTarget)
+        reactionInit(unitTarget)
       elseif event == "GROUP_ROSTER_UPDATE" then
-        for unit in pairs(watchUnitChange.unitIdToGUID) do
-          if Private.multiUnitUnits.raid[unit] or Private.multiUnitUnits.group[unit] then
+        for unit in pairs(Private.multiUnitUnits.group) do
```

Comment:
We can restrict ourselves to checking just Private.multiUnitUnits.group here (which is already a lot, but that can't be helped)